src/entities/Resource.js

```diff
@@ -113,4 +123,46 @@ export default class Resource {
   getTags() {
     return this.tags;
   }
+
+  /**
+   * Highlight all appearances of keyword in item.
+   * E.x: the keyword 'sa' will be highlighted 2 times in string 'saritasa/api'
+   *
+   * @param {string} keyword - Search keyword
+   */
+  addHighlight(keyword) {
```

Comment:
I think _addHighlight_ method not in the zone of responsibility **Resource** entity. May be we should create specific service for this ? Just example: Right now we have only one page with only 2 technology stack, what if tomorrow we add feature with choose  stack on separate page and each stack should has different highlighting ? But if we create specific service for this like **highlighterService** with method **addHighlight**  which would be receive technologyStacks and word to highlight it could be more flexible in the future. Also when in this case representation layer mixed with business-logic layer. What do you think ?

Reply:
Oh Resource is an entity. I thought that Resource is a component. This is my mistake.
I agree. We should move `addHighlight()` to a different place.